src/numpy-stubs/_typing/_array_like.pyi

```diff
@@ -59,6 +59,7 @@ _ArrayLikeFloat_co: TypeAlias = _DualArrayLike[np.dtype[np.floating | np.integer
 _ArrayLikeComplex_co: TypeAlias = _DualArrayLike[np.dtype[np.bool | np.number], complex]
 _ArrayLikeNumber_co: TypeAlias = _ArrayLikeComplex_co
 _ArrayLikeTD64_co: TypeAlias = _DualArrayLike[np.dtype[np.timedelta64 | np.integer | np.bool], int]
+_ArrayLikeTD64: TypeAlias = _DualArrayLike[np.dtype[np.timedelta64], np.timedelta64]
```

Comment:
`np.generic` and its subtypes are also assignable to `_SupportsArray`, so this is equivalent to `_ArrayLike[np.timedelta64]`.